model/task/task.go

```diff
@@ -664,18 +664,27 @@ func (t *Task) AllDependenciesSatisfied(cache map[string]Task) (bool, error) {
 }
 
 // HasFailedTests returns true if the task had any failed tests.
-func (t *Task) HasFailedTests() bool {
-	for _, test := range t.LocalTestResults {
-		if test.Status == evergreen.TestFailedStatus {
-			return true
+func (t *Task) HasFailedTests() (bool, error) {
+	// Check cedar flags before populating test results to avoid
+	// unnecessarily downloading results from cedar.
```

Comment:
Since `PopulateTestResults` could load from either Cedar or the DB, I would phrase this more like "avoid unnecessarily fetching test results".